### PR TITLE
Get VS 2015 to use const int definitions

### PR DESCRIPTION
**Pull request by randomascii**
VC++ up to VS 2015 RTM does not require explicit storage allocation for
static const integers declared in classes. VS 2015 Update 1 requires
these storage definitions in some cases. It's unclear exactly what
cases - simple tests work with and without the explicit storage
allocation.

Many previous versions of VC++ have theoretically *allowed* a
definition to supply storage, but tests on VC++ 2013 show that this
doesn't actually work correctly - it leads to duplicate definition
errors in Chromium. So, the change is scoped to VS 2015 only.

This change also updates the generated files to match the new code
generator.

TL;DR - this change is necessary in order for Chromium to build with
VS 2015 Update 1.

### CODE REVIEW

**Comment by randomascii** [comment]
This is to fix issue #917

**Comment by xfxyjwf** [comment]
Thanks Bruce! I'll merge this PR once the tests are done.

**Comment by randomascii** [comment]
All checks have passed.
